README.md

```diff
@@ -1,18 +1,36 @@
 # Candid
 
 ![](https://github.com/dfinity/candid/workflows/Rust/badge.svg)
-[![Crates.io](https://img.shields.io/crates/v/candid.svg)](https://crates.io/crates/candid)
-[![Documentation](https://docs.rs/candid/badge.svg)](https://docs.rs/candid)
 
 
-[Candid](IDL.md) is an interface description language (IDL) for interacting with _canisters_ (also known as _services_ or _actors_) running on the Internet Computer.
+[Candid](spec/Candid.md) is an interface description language (IDL) for interacting with _canisters_ (also known as _services_ or _actors_) running on the Internet Computer. It provides a language-independent description of canister interfaces and
+the data they exchange, with type safety and extensibility.
 
-This repository hosts the following crates:
+## Specifications
 
-- `candid`, a serialization/deserialization library for Candid. You can seamlessly convert between Rust values and Candid in both binary and text format.
-- `candid_derive`, an internal crate to convert Rust data types to Candid types. This crate should be considered as an implementation detail, and not be used directly, only via the `candid` crate.
+The [spec](spec/) directory contains Candid specifications, including the language specification, soundness proof and integration with other languages.
 
-# Contribution
+## Implementations
+
+Candid supports several different programming languages. 
+This repository contains some of the implementations developed by DFINITY.
+
+* [Rust](rust/): A serialization library based on Serde, and a compiler for generating bindings for other languages.
+* Motoko: Compiler support for importing/export Candid files.
+* JavaScript: As part of the [JavaScript User Library](https://www.npmjs.com/package/@dfinity/agent), we provide a library for serialization of native JavaScript values, and a visitor class for extending Candid for building generic tools such as UI and random testing.
+* More to come...
+
+A list of community maintained Candid libraries:
+
+* [Haskell](https://github.com/nomeata/haskell-candid)
+* [Elm](https://github.com/chenyan2002/ic-elm/)
+* [Kotlin](https://github.com/seniorjoinu/candid-kt)
```

Comment:
Oh wow, I didn't konw about this. This makes providing candid test data even more relevant (even a simple one).